maggy/ablation/ablator/loco.py

```diff
@@ -0,0 +1,186 @@
+from maggy.ablation.ablator import AbstractAblator
+from maggy.ablation.ablationstudy import AblationStudy
+from hops import featurestore
+from hops import hdfs as hopshdfs
+import tensorflow as tf
+from maggy.trial import Trial
+import json
+
+
+class LOCO(AbstractAblator):
+
+    def __init__(self, ablation_study, final_store):
+        super().__init__(ablation_study, final_store)
+        self.base_dataset_generator = self.get_dataset_generator(ablated_feature=None)
+
+    def get_number_of_trials(self):
+        # plus one is for the base (reference) trial with all the components
+        return len(self.ablation_study.features.included_features) + \
+            len(self.ablation_study.model.layers.included_layers) +  \
+            len(self.ablation_study.model.layers.included_groups) + 1
+
+    def get_dataset_generator(self, ablated_feature=None, dataset_type='tfrecord', shuffle_buffer_size=10000):
+
+        if self.ablation_study.custom_dataset_generator:
+            pass
+        else:
+            training_dataset_name = self.ablation_study.hops_training_dataset_name
+            training_dataset_version = self.ablation_study.hops_training_dataset_version
+            label_name = self.ablation_study.label_name
+
+            if dataset_type == 'tfrecord':
+
+                def create_tf_dataset(num_epochs, batch_size):
```

Comment:
Will this work if the dataset_type is not tfrecord? Maybe throw an "NotSupportedException" if the user pass another dataset type

Reply:
In such cases the users have to provide their own dataset generator functions (for now). Added a custom exception, and it will be raised there for now.
https://github.com/logicalclocks/maggy/pull/27/commits/6f2ed1a2dd3465c4ad54974a6e1e32a6ab797be7

Reply:
Ok, good!